TunnelKit/Sources/Protocols/OpenVPN/AppExtension/NEUDPLink.swift

```diff
@@ -58,28 +61,50 @@ class NEUDPLink: LinkInterface {
             guard let _ = self else {
                 return
             }
+            var packetsToUse: [Data]?
+            if self!.xorMask == 0 || packets == nil {
+                packetsToUse = packets
+            } else {
+                packetsToUse = packets!.map({ (packet) -> Data in
+                    return Data(bytes: packet.map{$0 ^ self!.xorMask}, count: packet.count)
+                })
+            }
```

Comment:
This code in dangerous. Replace _ in 61 and use it rather than `self!` for safety. Also, `packets` in the else block may be nil if `xorMask != 0`.